apps/rush-lib/src/logic/buildCache/ProjectBuildCache.ts

```diff
@@ -194,61 +229,103 @@ export class ProjectBuildCache {
     }
 
     terminal.writeVerboseLine(`Caching build output folders: ${filteredOutputFolders.join(', ')}`);
-    let encounteredTarErrors: boolean = false;
-    const tarStream: stream.Readable = tar.create(
-      {
-        gzip: true,
-        portable: true,
-        strict: true,
-        cwd: projectFolderPath,
-        filter: (tarPath: string, stat: tar.FileStat) => {
-          const tempStats: fs.Stats = new fs.Stats();
-          tempStats.mode = stat.mode;
-          if (tempStats.isSymbolicLink()) {
-            terminal.writeError(`Unable to include "${tarPath}" in build cache. It is a symbolic link.`);
-            encounteredTarErrors = true;
-            return false;
-          } else {
-            return true;
-          }
+    const tarExecutablePath: string | undefined = ProjectBuildCache._tarExecutablePath;
+    let localCacheEntryPath: string | undefined;
+
+    if (tarExecutablePath) {
+      const tempLocalCacheEntryPath: string = this._localBuildCacheProvider.getCacheEntryPath(cacheId);
+      const childProcess: ChildProcess = Executable.spawn(
+        tarExecutablePath,
+        ['-c', '-f', tempLocalCacheEntryPath, '-z', ...filteredOutputFolders],
```

Comment:
I highly recommend passing `--hard-dereference`, though maybe it doesn't matter if the tars stay on the local filesystem.

Reply:
That option isn't supported on the version of `tar` that comes with Windows 10.

Reply:
We might need to do platform detection and if on *nix, add the flag, because the hardlink optimizations can cause us problems. We experienced this a while back with `npm pack` operating on files, some of which were hardlinks.